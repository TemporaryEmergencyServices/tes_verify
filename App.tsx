import { StatusBar } from 'expo-status-bar';
import React from 'react';
import { SafeAreaProvider } from 'react-native-safe-area-context';

import useCachedResources from './hooks/useCachedResources';
import useColorScheme from './hooks/useColorScheme';
import Navigation from './navigation';


//The following are for redux
import {createStore, applyMiddleware} from 'redux'
import {Provider} from 'react-redux'
import thunkMiddleware from 'redux-thunk'
// import reducer from './reducers'
import rootReducer from './reducers'

const middleware = applyMiddleware(thunkMiddleware)
export const store = createStore(rootReducer, middleware)
// const store = createStore(rootReducer)
//End redux

export default function App() {
  const isLoadingComplete = useCachedResources();
  const colorScheme = useColorScheme();

  if (!isLoadingComplete) {
    return null;
  } else {
    return (
      //Provider wrapper and store area for redux
      <Provider store={store}>
        <SafeAreaProvider>
            <Navigation colorScheme={colorScheme} />
            <StatusBar />
        </SafeAreaProvider>
      </Provider>
    );
  }
}