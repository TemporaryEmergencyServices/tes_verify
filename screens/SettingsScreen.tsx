import * as React from 'react';
import { StyleSheet, Button, Alert, TouchableOpacity } from 'react-native';
import firebase from '../firebase.js'

import EditScreenInfo from '../components/EditScreenInfo';
import { Text, View } from '../components/Themed';
import { useSelector, useDispatch, RootStateOrAny } from 'react-redux'
import { logout } from '../actions'


export default function SettingsScreen({ navigation }) {
  const user = useSelector((state: RootStateOrAny) => state.user)
  const dispatch = useDispatch()

  const goToApply = () => navigation.push('ApplyScreen')
  const goToSignIn = () => navigation.replace('SignInScreen')
  const goToApprove= () => navigation.navigate('ManagerApproveApplicationScreen')

  const handleApply = () => {goToApply()}
  const handleApprove = () => {goToApprove()}
  const handleLogout = () => {
    firebase.auth().signOut()
    .then(() => dispatch(logout()))
    .then(goToSignIn)
    .catch(error => {
      Alert.alert(
        "Error",
        error.message,
        [
          { text: "OK", onPress: () => console.log("OK Pressed") }
        ],
        { cancelable: false }
      );
    })
  }

  return (
    
    <View style={styles.container}>
      
      <Text style={styles.title}> {user.email}</Text>
      <TouchableOpacity style={styles.signOutBtn} onPress={handleLogout}>
        <Text style={styles.signOutText}>Sign Out</Text>
      </TouchableOpacity>  

      <TouchableOpacity style={styles.signOutBtn} onPress = {handleApply}>
        <Text style={styles.signOutText}>Apply to be a volunteer</Text>
      </TouchableOpacity> 
      <TouchableOpacity style={styles.signOutBtn} onPress = {handleApprove}>
        <Text style={styles.signOutText}>Approve volunteer applications</Text>
      </TouchableOpacity> 
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  title: {
    fontSize: 20,
    fontWeight: 'bold',
    textAlign: 'center',
    color: "#1C5A7D"
  },
  separator: {
    marginVertical: 30,
    height: 1,
    width: '80%',
  },
  signOutBtn:{
    width:"80%",
    backgroundColor:"#E11383",
    borderRadius:25,
    height:50,
    alignItems:"center",
    justifyContent:"center",
    marginTop:40,
    marginBottom:10
  },
  signOutText:{
    marginTop:10,
    color:"white",
    justifyContent: 'center',
    alignContent: 'center',
    fontWeight :'bold',
    fontSize: 18, 
    paddingBottom: 10
  }
});
