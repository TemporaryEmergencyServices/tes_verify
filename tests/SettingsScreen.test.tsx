import React from "react";
import renderer from "react-test-renderer";
import SettingsScreen from "../screens/SettingsScreen"
import { Provider } from 'react-redux'
import { store } from '../App'

jest.useFakeTimers()

describe("<SettingsScreen />", () => {
    it('renders correctly', () => {
        const navigation = {}
        const tree = renderer.create(<Provider store={store} ><SettingsScreen navigation={navigation} /></Provider>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});