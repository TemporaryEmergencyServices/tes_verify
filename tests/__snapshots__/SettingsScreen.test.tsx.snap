// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SettingsScreen /> renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "color": "#1C5A7D",
          "fontSize": 20,
          "fontWeight": "bold",
          "textAlign": "center",
        },
      ]
    }
  >
     
  </Text>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1C5A7D",
        "borderRadius": 25,
        "height": 50,
        "justifyContent": "center",
        "marginBottom": 10,
        "marginTop": 40,
        "opacity": 1,
        "width": "80%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
          },
          Object {
            "alignContent": "center",
            "color": "white",
            "fontSize": 18,
            "fontWeight": "bold",
            "justifyContent": "center",
            "marginTop": 10,
            "paddingBottom": 10,
          },
        ]
      }
    >
      Apply to be a volunteer
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#1C5A7D",
        "borderRadius": 25,
        "height": 50,
        "justifyContent": "center",
        "marginBottom": 10,
        "marginTop": 40,
        "opacity": 1,
        "width": "80%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
          },
          Object {
            "alignContent": "center",
            "color": "white",
            "fontSize": 18,
            "fontWeight": "bold",
            "justifyContent": "center",
            "marginTop": 10,
            "paddingBottom": 10,
          },
        ]
      }
    >
      Approve volunteer applications
    </Text>
  </View>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#E11383",
        "borderRadius": 25,
        "height": 50,
        "justifyContent": "center",
        "marginBottom": 10,
        "marginTop": 40,
        "opacity": 1,
        "width": "80%",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#000",
          },
          Object {
            "alignContent": "center",
            "color": "white",
            "fontSize": 18,
            "fontWeight": "bold",
            "justifyContent": "center",
            "marginTop": 10,
            "paddingBottom": 10,
          },
        ]
      }
    >
      Sign Out
    </Text>
  </View>
</View>
`;
